fe/fe-core/src/main/java/org/apache/doris/planner/Planner.java

```diff
@@ -266,6 +271,57 @@ public void createPlanFragments(StatementBase statement, Analyzer analyzer, TQue
         }
     }
 
+    private class FindTuplePredicate implements com.google.common.base.Predicate<PlanNode> {
+        private final TupleId tupleId;
+
+        FindTuplePredicate(TupleId id) {
+            tupleId = id;
+        }
+
+        @Override
+        public boolean apply(PlanNode arg) {
+            return arg.getTupleIds().size() == 1 && arg.getTupleIds().get(0) == tupleId
```

Comment:
I understand that filtering conjuncts to be empty is to avoid invalid filtering in some primary-foreign key joins, because the key values ​​on the left and right sides are the same at this time; limiting the number of Tuples to 1 skips the child join node.

It is recommended to add a note here, it is not very well understood;

Reply:
Note is added, pls see the new commit